test/controllers/api/v2/realms_controller_test.rb

```diff
@@ -53,42 +53,42 @@ class Api::V2::RealmsControllerTest < ActionController::TestCase
     refute Realm.unscoped.find_by_id(realm['id'])
   end
 
-  # test that taxonomy scope works for api for realms
-  def setup
-    taxonomies(:location1).realm_ids = [realms(:myrealm).id, realms(:yourrealm).id]
-    taxonomies(:organization1).realm_ids = [realms(:myrealm).id]
-  end
+  context 'taxonomy scope' do
+    let (:myrealm) { realms(:myrealm) }
+    let (:yourrealm) { realms(:yourrealm) }
+    let (:loc) { FactoryBot.create(:location, realms: [myrealm, yourrealm]) }
+    let (:org) { FactoryBot.create(:organization, realms: [myrealm]) }
 
-  test "should get realms for location only" do
-    get :index, params: { :location_id => taxonomies(:location1).id }
-    assert_response :success
-    assert_equal taxonomies(:location1).realms.length, assigns(:realms).length
-    assert_equal assigns(:realms), taxonomies(:location1).realms
-  end
+    test "should get realms for location only" do
+      get :index, params: { :location_id => loc.id }
+      assert_response :success
+      assert_equal loc.realms.length, assigns(:realms).length
+      assert_equal assigns(:realms).sort, loc.realms.sort
```

Comment:
This is the line that I believe will fix the transient failures we're seeing. It looks as if in certain cases one of these is sorted the other way, so the equal comparison would fail.